src/Evolution/Systems/Cce/Actions/BoundaryComputeAndSendToEvolution.hpp

```diff
@@ -130,6 +130,114 @@ struct BoundaryComputeAndSendToEvolution<H5WorldtubeBoundary<Metavariables>,
+ * managing the file buffer, interpolating to the desired time point, and
+ * compute the Bondi and Klein-Gordon quantities on the boundary. Once readied,
+ * it sends each tensor or scalar from the the full `Variables<typename
+ * Metavariables::cce_boundary_communication_tags>` or `Variables<typename
+ * Metavariables::klein_gordon_boundary_communication_tags>` back to the
+ * `EvolutionComponent`
+ *
+ * Uses:
+ * - DataBox:
+ *  - `Tags::H5WorldtubeBoundaryDataManager`
+ *  - `Tags::KleinGordonH5WorldtubeBoundaryDataManager`
+ *
+ * \ref DataBoxGroup changes:
+ * - Adds: nothing
+ * - Removes: nothing
+ * - Modifies:
+ *   - `Tags::Variables<typename
+ * Metavariables::cce_boundary_communication_tags>` (every tensor)
+ *   - `Tags::Variables<typename
+ * Metavariables::klein_gordon_boundary_communication_tags>` (every scalar)
+ */
+template <typename Metavariables, typename EvolutionComponent>
+struct BoundaryComputeAndSendToEvolution<
+    KleinGordonH5WorldtubeBoundary<Metavariables>, EvolutionComponent> {
+  template <typename ParallelComponent, typename... DbTags, typename ArrayIndex>
+  static void apply(db::DataBox<tmpl::list<DbTags...>>& box,
+                    Parallel::GlobalCache<Metavariables>& cache,
+                    const ArrayIndex& /*array_index*/, const TimeStepId& time) {
+    auto hdf5_lock = Parallel::local_branch(
+                         Parallel::get_parallel_component<
+                             observers::ObserverWriter<Metavariables>>(cache))
+                         ->template local_synchronous_action<
+                             observers::Actions::GetLockPointer<
+                                 observers::Tags::H5FileLock>>();
+    bool successfully_populated = false;
+    db::mutate<Tags::H5WorldtubeBoundaryDataManager,
+               ::Tags::Variables<
+                   typename Metavariables::cce_boundary_communication_tags>>(
+        [&successfully_populated, &time, &hdf5_lock](
+            const gsl::not_null<std::unique_ptr<Cce::WorldtubeDataManager<
+                Tags::characteristic_worldtube_boundary_tags<
+                    Tags::BoundaryValue>>>*>
+                worldtube_data_manager,
+            const gsl::not_null<Variables<
+                typename Metavariables::cce_boundary_communication_tags>*>
+                boundary_variables) {
+          successfully_populated =
+              (*worldtube_data_manager)
+                  ->populate_hypersurface_boundary_data(
+                      boundary_variables, time.substep_time(), hdf5_lock);
+        },
+        make_not_null(&box));
+    if (not successfully_populated) {
+      ERROR(
+          "Insufficient tensor boundary data to proceed, exiting early at "
+          "time " +
+          std::to_string(time.substep_time()));
+    }
+
+    successfully_populated = false;
+    db::mutate<
+        Tags::KleinGordonH5WorldtubeBoundaryDataManager,
+        ::Tags::Variables<
```

Comment:
You could combine the `mutate`s here into one and just ensure that both are successful rather than having two `mutate`s.

Reply:
Good idea. Done!